pages/common/xargs.md

```diff
@@ -1,19 +1,20 @@
 # xargs
 
-> Execute a command with piped arguments.
+> Execute a command with piped arguments coming from another command, a file, etc.
+> The input is treated as a single block of text and split into separate arguments on spaces, tabs, newlines and end-of-file.
 
-- Main use:
+- Main usage pattern:
 
-`{{arguments}} | xargs {{command}}`
+`{{arguments_source}} | xargs {{command}}`
 
-- Specific example: delete all files that start with 'M':
+- Delete all files with a `.backup` extension:
 
-`find . -name 'M*' | xargs rm`
+`{{find . -name '*.backup'}} | xargs {{rm -v}}`
 
-- Handle whitespace in arguments:
+- Use the NUL (`\0`) character as separator (useful to split the input on newlines but not on spaces):
```

Comment:
I'm inclined to disagree. The "what" is important, but if we don't emphasise the "how" (indeed, "why") it works, we risk defrauding our mission to allow people to *learn* the syntax of commands by example as opposed to memorizing obscure formulas. This was the problem in the previous description which made it hard for me to get a grasp on what was going on until I read the manpage -- which, by the way, describes the -0 option as "Change xargs to expect NUL (``\0'') characters as separators, instead of spaces and newlines." -- which is literally what it does, and is quite accessible already, IMO.

That said, it's not that the description you propose fails to explain the how, but I fear that the two phrases "split input in newlines" and "using the NUL character as separator" may make the understanding harder, since \n and \0 are different characters and xargs is explicitly using the latter -- the conversion is done by the user, but xargs is blind to that.

I'm not sure I'm being clear. Let me know if I'm rambling or if I need to clarify some of my points.